accepted/0000-sbom-command.md

```diff
@@ -0,0 +1,461 @@
+# SBOM Generation for npm Projects
+
+## Summary
+
+Update the npm CLI with a new command which will generate a Software Bill of Materials (SBOM) containing an inventory of the current project's dependencies. Users will have the option to generate an SBOM conforming to either the [Software Package Data Exchange (SPDX](https://spdx.github.io/spdx-spec/v2.3/)) or [CycloneDX](https://cyclonedx.org/specification/overview/) specifications.
+
+
+## Motivation
+
+Finding and remediating vulnerabilities in open source projects is a critical component of securing the software supply chain. However, this requires that enterprises understand what OSS components are used across their infrastructure and applications. When new vulnerabilities are discovered, they need a complete inventory of their software assets in order to properly assess their exposure. Knowing about the critical [Log4j](https://www.cisa.gov/news-events/cybersecurity-advisories/aa21-356a#main) vulnerability doesn’t do you any good unless you can also pinpoint where in your organization you’re running the vulnerable code.
+
+SBOMs help to solve this problem by providing a standardized way to document the components that comprise a software application. A proper SBOM should tell you exactly which packages you have deployed and which versions of those packages you are using.
+
+Beyond the security benefit there may be a regulatory requirement to provide SBOMs in some sectors. In response to some recent, high-visibility attacks, The White House has issued an [Executive Order](https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/) which specifically includes directives which would make SBOMs a requirement for any vendor selling to the federal government.
+
+Adding SBOM generation to the tool which many developers are already using to manage their project dependencies eliminates any friction which may come from having to adopt/learn a separate tool. 
+
+
+## Detailed Explanation
+
+A new `sbom` command will be added to the npm CLI which will generate an SBOM for the current project. The SBOM will use the current project as the root and enumerate all of its dependencies (similar to the way `npm-ls` does) in one of the two supported SBOM formats. See the 
+[Example SBOMs](#example-sboms) section for sample CycloneDX and SPDX SBOM documents.
+
+Supported command options: 
+
```

Comment:
missing option:

SBOM may contain whether the system under analysis is an application or of other type.
see the examples in the RFC.
maybe the project under analysis is not an application, but an encapsulated library with bundled dependencies?
or it is a firmware for an embedded JavaScript (like [elk](https://github.com/cesanta/elk))
this is why [@cyclonedx/cyclonedx-npm](https://www.npmjs.com/package/%40cyclonedx/cyclonedx-npm) has an option for that:
```text
--mc-type <type>          Type of the main component.
                            (choices: "application", "firmware", "library", default: "application")
```

Reply:
Looks like this has good overlap with the SPDX [primaryPackagePurpose](https://spdx.github.io/spdx-spec/v2.3/package-information/#724-primary-package-purpose-field) field

Reply:
This is indeed the "purpose" of the primaryPackagePurpose.

Reply:
Added an option to capture the "type" which will be used to populate either the CDX `type` field or the SPDX `primaryPackagePurpose` field.